apollo3.cpp

```diff
@@ -0,0 +1,135 @@
+// This provides the functionality for Apollo3 devices.
+
+#if defined(AM_PART_APOLLO3)
+
+#include <ap3_types.h>
+#include <am_hal_gpio.h>
+
+#include "Adafruit_NeoPixel.h"
+
+// The timing method used to control the NeoPixels
+// TODO: Implement something better (interrupts, DMA, etc)
+#define PIN_METHOD_FAST_GPIO
+
+/*!
+  @brief   Unset the NeoPixel output pin number.
+  @param   p  Arduino pin number (-1 = no pin).
+*/
+void Adafruit_NeoPixel::apollo3UnsetPin(uint16_t pin) {
+#if defined(PIN_METHOD_FAST_GPIO)
+  // Unconfigure the pin for Fast GPIO.
+  am_hal_gpio_fastgpio_disable(pin);
+#endif
+}
+
+/*!
+  @brief   Set the NeoPixel output pin number.
+  @param   p  Arduino pin number (-1 = no pin).
+*/
+void Adafruit_NeoPixel::apollo3SetPin(uint16_t pin) {
```

Comment:
As written this function will likely confuse users b/c NeoPixel output may be mapped incorrectly. The input parameter here is the Arduino pin while the calls to ```am_hal_gpio_xxx``` expect the Apollo3 pad number. The mapping is dependent on the board variant and uses ```ap3_gpio_pad_t ap3_gpio_pin2pad(ap3_gpio_pin_t pin);``` which is available wherever "Arduino.h" is included. 

This applies to any/all cases in this file where the pin is passed directly into an Ambiq HAL function.

Reply:
I decided to apply the pin-to-Apollo3 pad mapping in the Adafruit_NeoPixel.cpp `setPin` function.  That way, the mapping only needs to be performed a single time instead of every Ambiq HAL function, and we can avoid that extra function call in every `show()`.

Reply:
Now that the pin-to-pad mapping is implemented, I'm not happy with how I named some of the functions.  Going to post another update with more appropriate names.